pkg/datapath/maps/map.go

```diff
@@ -34,52 +34,73 @@ import (
 
 var (
 	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "datapath-maps")
+
+	globalSweeper = newMapSweeper(&realGarbageWalker{})
 )
 
-// CollectStaleMapGarbage cleans up stale content in the BPF maps from the
-// datapath.
-func CollectStaleMapGarbage() {
-	if option.Config.DryMode {
-		return
-	}
-	walker := func(path string, _ os.FileInfo, _ error) error {
-		return staleMapWalker(path)
-	}
+// garbageWalker checks against its list of the current endpoints to determine
+// whether map paths should be removed, and implements map removal.
+type garbageWalker interface {
+	shouldRemove(id uint16) bool
```

Comment:
endpointID? what's the ID here? Why is this ID a `uint16` and the ID on the ID bellow is `uint32`?

Reply:
Honestly I was just refactoring the existing code and didn't think about it. Yes, it's an endpoint and I'll make this more clear.

I'm not sure why the existing code treats the endpoint id as a different type in different places, probably a detail of the bpf map implementation and the function signatures provided there. I'll take a look.